example/src/App.js

```diff
@@ -18,9 +18,12 @@ export function App() {
     <NavigationContainer onStateChange={Instabug.onStateChange}>
       <Tab.Navigator
         screenOptions={({ route }) => ({
-          tabBarIcon: props => {
+          // eslint-disable-next-line react/no-unstable-nested-components
+          tabBarIcon: (props) => {
```

Comment:
Extracting `TabBarIcon` to its own component still will lead into the same issue, since we need to pass both the `route` and `props` to it as follows:

```js
tabBarIcon: (props) => <TabBarIcon route={route} {...props} />,
```

Do you have another suggestion in mind?

Reply:
What about a higher-order component (HOC)?
```js
function createTabBarIcon(route) {
  return (props) => {
    // the tab bar icon logic in here
  }
}
```

then we can use it like this:

```js
tabBarIcon: createTabBarIcon(route),
```

and the props will be passed automatically since it returns a regular React component.